soh/soh/Enhancements/game-interactor/GameInteractor.h

```diff
@@ -221,6 +221,7 @@ class GameInteractor {
 
     DEFINE_HOOK(OnFileDropped, void(std::string filePath));
     DEFINE_HOOK(OnAssetAltChange, void());
+    DEFINE_HOOK(OnPauseMenu, void());
```

Comment:
I think this maybe makes more sense as `OnKaleidoUpdate`

I also assume that this means `OnGameFrameUpdate` does not run with kaleido open? So you couldn't just also use `OnGameFrameUpdate` and instead just check some sort of pauseCtx value?

Reply:
Nah, OnGameFrameUpdate does run with Kaleido open. You're right, I could of just checked if pauseCtx->pageIndex == PAUSE_QUEST. However, on my original PR for this jbodner09 like the idea of having a hook specific to the pause menu for custom item mods. That comment is [here](https://github.com/HarbourMasters/Shipwright/pull/3212#issuecomment-1724958636). That's why I kept it in, but it's up to yall. It's not necessary for my use case so I could remove it.

Reply:
On second thought, `OnGameFrameUpdate` is not something we want to use often, so this may be better after all

Reply:
I agree. I tried to figure out a way around using it for PauseWarp_Execute, but couldn't figure out a easy way to do it. I've made the checks pretty specific so it shouldn't cause any issue anyway.

Reply:
Yea should be fine, I still stand by my original rename suggestion though 👍

Reply:
Fine by me and I just changed it in the latest commit.